### DISCUSSION TITLE
[GSOC] Interested in project "Linux command line (sh/Perl/Python) interface to Google groups"

### DISCUSSION THREAD

**Discussion by rsnj5** [unicode-org/icu4x#4727] [Q&A]
Hello everyone!

I am Sanju Raj, an undergraduate student from India majoring in Electronics and Computer Science. I have completed various projects in Python and am familiar with bash scripting. Additionally, I have worked with Google Groups' API for managing a club's group at my university. Though I am relatively new to open source, I believe I possess the necessary skills to successfully tackle this project. Currently, I am in the process of familiarizing myself with LGL. I would greatly appreciate it if anyone could provide pointers on how to get started with this project. Furthermore, since this project differs significantly from the core code base of Unicode, could anyone suggest where I can begin contributing?

Thank you!

**Comment by rsnj5** [comment]
Also I am not sure if this is the right discussion tab to this project, since this was linked to the contributor guidance I posted this here. If this is not the right place, where can I get the guidance for this project?

**Reply by sffc** [reply]
Hi, @rsnj5!

@rickmcgowan would be your mentor for this project and can share more details on the project.

**Reply by sffc** [reply]
Also CC @elfkuzco who posted in #4680:

> Hi @sffc , I am interested in working on "Linux command line (sh/Perl/Python) interface to Google groups" and have an understanding of the project requirements. Looking at the Google API for groups, I can develop a Python command line that will take inputs either through a file or directly on the command line. However, it is mentioned that " it would be much better if we could keep many/most groups constantly in sync by using programs to do automatic updates based on daily LGL reports about our constituents.". I would like to enquire the format of the LGL reports and any further guides on how to begin.

**Comment by rickmcgowan** [comment]
Hello,

Your inquiry was forwarded to me regarding the GSoC project for interfacing to Google Groups. I can try to answer any questions you might have about this project. There should be a lot of basic information in the GSoC description, but please feel free to ask me anything about the project, specifically what may not be covered there.

As for one question: LGL reports are in CSV, TSV, or XLSX format, and can be inter-converted. Not really looking for code to parse those reports, rather developing an interface to modify/update Google group contents, typically from within Linux processes that are already parsing those reports for other purposes. Please feel free to DM me and/or send email directly.

Regards,

**Reply by rsnj5** [reply]
Hi @rickmcgowan,
Thanks for clarifying the report part. Another question was regarding the relevant repo to contribute. As of now, I am working on basic prototype of this project to understand Google APIs better. I am working on a script to manage the various Google groups in a school setting. For now, I have assumed three roles, principal/head, teachers, parents, and students. Would you suggest something else to showcase my competence?

**Reply by sffc** [reply]
Dear @rsnj5: This is just a reminder to make sure your application is submitted through the tool by the deadline, April 2 at 18:00 UTC. You are also welcome to email gsoc@unicode.org with any questions.

**Reply by rsnj5** [reply]
Hello @sffc @rickmcgowan I have submitted a proposal at the Google site. Please have a look at it and suggest possible improvements
